src/Onboarding/Routes/CurrencyRoute.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Give\Onboarding\Routes;
+
+use WP_REST_Request;
+use Give\API\RestRoute;
+use Give\Onboarding\SettingsRepository;
+
+class CurrencyRoute implements RestRoute {
+
+	protected $endpoint = 'onboarding/settings/currency';
```

Comment:
This "extends" the `onboarding/settings/{setting}" endpoint, providing specific handling for the currency setting with has side effects.